src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.GenerateAssemblyInfo.targets

```diff
@@ -84,7 +81,35 @@ Copyright (c) .NET Foundation. All rights reserved.
         <_Parameter1>$(NeutralLanguage)</_Parameter1>
       </AssemblyAttribute>
     </ItemGroup>
+  </Target>
+
+  <!-- 
+    To allow version changes to be respected on incremental builds (e.g. through CLI parameters),
+    create a hash of all assembly attributes so that the lock file name will change with the calculated
+    assembly attribute values and msbuild will then execute CoreGenerateAssembly to generate a new file.
+  -->
+  <Target Name="_CreateGeneratedAssemblyInfoInputsCacheFile"
+          DependsOnTargets="_CalculateAssemblyAttributes"
+          BeforeTargets="CoreGenerateAssemblyInfo">
```

Comment:
Does the `DependsOnTargets="A;B;C"` set the build dependency of "b depends on a"? 

Option: Merge `_CreateGeneratedAssemblyInfoInputsCacheFile` and `_CalculateAssemblyAttributes` into `GetAssemblyAttributes `, add `DependsOnTargets="GetAssemblyAttributes"` to `CoreGenerateAssemblyInfo `.

That would merge both concerns of emitting the items and writing the cache but leave nicely named public targets. @nguerrera your call